File: app/src/main/java/com/themaker/fshmo/klassikaplus/data/mappers/DbToDomainItemMapper.java
package com.themaker.fshmo.klassikaplus.data.mappers;

import android.util.Log;
import com.themaker.fshmo.klassikaplus.data.domain.Item;
import com.themaker.fshmo.klassikaplus.data.persistence.model.DbItem;

public class DbToDomainItemMapper extends Mapping<DbItem, Item> {

    private static final String TAG = DbToDomainItemMapper.class.getName();

    @Override
    public Item map(DbItem dbItem) {
        Item item = new Item();
        item.setId(dbItem.getId());
        item.setExtId(dbItem.getExtId());
        item.setIcon(dbItem.getIcon());
        item.setPrice(dbItem.getPrice());
        item.setName(dbItem.getName());
        item.setNovelty(dbItem.getNovelty());
        item.setPageAlias(dbItem.getPageAlias());
        item.setCategory(dbItem.getCategory());
        Log.d(TAG, "map: Item parsed: " + item.toString());
        return item;
    }
}

File: app/src/main/java/com/themaker/fshmo/klassikaplus/data/mappers/DbToDomainItemMapper.kt
package com.themaker.fshmo.klassikaplus.data.mappers

import android.util.Log
import com.themaker.fshmo.klassikaplus.data.domain.Item
import com.themaker.fshmo.klassikaplus.data.persistence.model.DbItem

class DbToDomainItemMapper : Mapping<DbItem, Item>() {

    override fun map(dbItem: DbItem): Item {
        val item = Item()
        item.id = dbItem.id
        item.setExtId(dbItem.extId)
        item.icon = dbItem.icon
        item.price = dbItem.price
        item.name = dbItem.name
        item.novelty = dbItem.novelty
        item.pageAlias = dbItem.pageAlias
        item.category = dbItem.category
        Log.d(TAG, "map: Item parsed: $item")
        return item
    }

    companion object {

        private val TAG = DbToDomainItemMapper::class.java.name
    }
}
